### PR TITLE
Remove "Reactive" dependency

### PR DESCRIPTION
**Pull request by jmccausland**
This PR should allow you to remove the "Reactive" dependency from the Pro project.  It was only using one class/method (Subject.Sample) to throttle the mouse movement handler to keep things perform-ant.  Sharing a class here that we created recently to do the same thing, enjoy.

Please be sure to test thoroughly before merging.

### CODE REVIEW

**Comment by jmccausland** [comment]
@csmoore please review and let me know if this is something that can be merged before our release on the 25th of this month.

**Comment by csmoore** [comment]
@jmccausland - many thanks for changes, I'll check with @lfunkhouser and @kgonzago to see if OK to do just before release

File: source/addins/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule.csproj [hunk 1]
```diff
@@ -57,22 +57,6 @@
   <ItemGroup>
     <Reference Include="System" />
     <Reference Include="System.Core" />
-    <Reference Include="System.Reactive.Core, Version=2.2.5.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\Dependencies\ReactiveExtensions\net45\System.Reactive.Core.dll</HintPath>
```
[the remark this answers is not in the document]
Comment (jmccausland) [reply source/addins/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule.csproj]: yep, was leaving that up to you guys

**Comment by kgonzago** [comment]
@jmccausland Thanks for the help.

@csmoore My suggestion is to merge after our release.

File: source/addins/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (csmoore) [reply source/addins/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule.csproj]: Remove those files in commit https://github.com/Esri/distance-direction-addin-dotnet/pull/640/commits/436b22890d28b451b21364b6c71a61d1f6baeefd in PR

**Comment by jmccausland** [comment]
@lfunkhouser @csmoore @kgonzago This was a simple replacement and our project uses the same throttle method.  Very low risk.

**Comment by csmoore** [comment]
Revert code to C# 5 for VS 2013 and created build at:  

Solutions\DefenseSolutions\DistanceAndDirection\Testing\jrm-remove-reactive-depends\15\

File: source/addins/DistanceAndDirectionLibrary/Helpers/DebounceDispatcher.cs [hunk 2]
```diff
@@ -39,7 +41,7 @@ public class DebounceDispatcher
             Dispatcher disp = null)
         {
             // kill pending timer and pending ticks
-            timer?.Stop();
+            timer.Stop();
```
Comment (jmccausland) [inline source/addins/DistanceAndDirectionLibrary/Helpers/DebounceDispatcher.cs]: you need to null check here before using timer.Stop, that's what the ?. does.  Null Propagation.

Comment (csmoore) [reply source/addins/DistanceAndDirectionLibrary/Helpers/DebounceDispatcher.cs]: Added in https://github.com/Esri/distance-direction-addin-dotnet/pull/640/commits/b9a03167923aedcff3caa85e8c2c24a98a09b609

**Comment by csmoore** [comment]
Created build `Solutions\DefenseSolutions\DistanceAndDirection\Testing\jrm-remove-reactive-depends\17\` to test.  It looks like this change only affects Pro so only needs testing there.

**Review by csmoore — APPROVED** [review]
Tested in Pro Addin (only app that uses these files/solution) and mouse feedback seemed smooth with no issues noticed.

Looks like this approach was never used in ArcMap - was there a reason (like not needed there)?

**Pull request merged**